### PR TITLE
Add NuGet.props file.

### PR DESCRIPTION
**Pull request by cristinamanum**
Add the NuGet.props  that will import the Directory.Packages.props for Central version management.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  Need to understand if there is necessary testing infrastructure for props validation.
Validation:  Manual testing

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: @rainersigwald I would appreciate your review for this file.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿<!--
+***********************************************************************************************
+Microsoft.Common.props
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Change the name.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: I don't think you need tools version, it doesn't do much here.

**Review by nkolev92 — COMMENTED** [review]
Have you tried out whether NuGet.exe can work with this?

Specifically, nuget.exe writes a targets file and run GenerateRestoreSpec on that targets file with a list of projects.

**Comment by cristinamanum** [comment]
For the comment "Have you tried out whether NuGet.exe can work with this?"

Tested with :
**nuget.exe** (with the props deployed to ***\Microsoft Visual Studio\Version\..\MSBuild\Current\Imports\Microsoft.Common.Props\ImportBefore)
**msbuild /t:restore** (with the props deployed to ***\Microsoft Visual Studio\Version\..\MSBuild\Current\Imports\Microsoft.Common.Props\ImportBefore)
**dotnet restore** (with the props deployed to (dotnet\sdk\version\Current\Imports\Microsoft.Common.props\ImportBefore)

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 2 above)
Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Updated.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 1 above)
Comment (rainersigwald) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]:
```suggestion
<Project>
```

Neither of these is required any more.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+-->
+
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- 
+        Import Directory.Packages.props.
+    -->
+  <PropertyGroup Condition="'$(DirectoryPackagesPropsPath)' == ''">
+    <_DirectoryPackagesPropsFile Condition="'$(_DirectoryPackagesPropsFile)' == ''">Directory.Packages.props</_DirectoryPackagesPropsFile>
+    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
+    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([System.IO.Path]::Combine('$(_DirectoryPackagesPropsBasePath)', '$(_DirectoryPackagesPropsFile)'))</DirectoryPackagesPropsPath>
```
Comment (rainersigwald) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]:
```suggestion
    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::GetPathOfFileAbove('$(_DirectoryPackagesPropsFile)'))</DirectoryPackagesPropsPath>
```

This intrinsic was added in 15.0.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- 
+        Import Directory.Packages.props.
+    -->
+  <PropertyGroup Condition="'$(DirectoryPackagesPropsPath)' == ''">
```
Comment (rainersigwald) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Can you add a condition that can disable the scan entirely? Something analogous to `$(ImportDirectoryBuildProps)`. Probably `$(ImportDirectoryPackagesProps)`.

Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Added

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 3 above)
Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Added like below. The suggestion did not load the file. 

``` xml
<DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::GetPathOfFileAbove('$(_DirectoryPackagesPropsFile)', '$(MSBuildProjectDirectory)'))</DirectoryPackagesPropsPath>

```

Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props]: Also removed the tools version (related with https://github.com/NuGet/NuGet.Client/pull/3219#discussion_r373739668 and https://github.com/NuGet/NuGet.Client/pull/3219#discussion_r375235241)

**Review by rainersigwald — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<!--
+***********************************************************************************************
+NuGet.props
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:9]: I know this is the correct one to use today, see https://github.com/NuGet/Home/issues/6518, but NuGet.targets is .NET Foundation. 

https://github.com/NuGet/NuGet.Client/blob/a746cd6a01dfd22187375ee9c5a0d9b98676020a/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L9

I am not blocking this issue on thsi, just wanted to make sure @karann-msft see this.

**Review by nkolev92 — APPROVED** [review]
🥇

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    <_DirectoryPackagesPropsFile Condition="'$(_DirectoryPackagesPropsFile)' == ''">Directory.Packages.props</_DirectoryPackagesPropsFile>
+    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
+    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::GetPathOfFileAbove('$(_DirectoryPackagesPropsFile)', '$(MSBuildProjectDirectory)'))</DirectoryPackagesPropsPath>
+  </PropertyGroup>
+
+  <Import Project="$(DirectoryPackagesPropsPath)" Condition="'$(ImportDirectoryPackagesProps)' == 'true' and '$(DirectoryPackagesPropsPath)' != '' and exists('$(DirectoryPackagesPropsPath)')"/>
+
+  <PropertyGroup Condition="'$(ImportDirectoryPackagesProps)' == 'true' and '$(DirectoryPackagesPropsPath)' != '' and exists('$(DirectoryPackagesPropsPath)')">
+    <CentralPackageVersionsFileImported>true</CentralPackageVersionsFileImported>
+  </PropertyGroup>
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:30]: Is there a benefit to having this second `PropertyGroup`?  Move the `CentralPackageVersionsFileImported` property to the first property group, which already has a `ImportDirectoryPackagesProps` condition, then just add the `exists` condition to the property itself.

**Review by zivkan — APPROVED** [review]

**Pull request merged**